server/endpoint/health/searcher/endpoint.go

```diff
@@ -77,23 +79,39 @@ func (e *Endpoint) Encoder() kithttp.EncodeResponseFunc {
 
 func (e *Endpoint) Endpoint() kitendpoint.Endpoint {
 	return func(ctx context.Context, request interface{}) (interface{}, error) {
-		var serviceRequest searcher.Request
-		{
-			clusterID, ok := request.(string)
-			if !ok {
-				return nil, microerror.Mask(badRequestError)
-			}
-			serviceRequest = searcher.Request{
-				ClusterID: clusterID,
-			}
+		clusterID, ok := request.(string)
+		if !ok {
+			return nil, microerror.Mask(badRequestError)
 		}
 
-		serviceResponse, err := e.service.Health.Searcher.Search(ctx, serviceRequest)
+		clusterRequest := cluster.Request{
+			ClusterID: clusterID,
+		}
+		clusterResponse, err := e.service.Cluster.Searcher.Search(ctx, clusterRequest)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+
+		nodeRequest := node.Request{
+			ClusterID: clusterID,
+			Endpoint:  clusterResponse.Endpoint,
+		}
+		nodeResponse, err := e.service.Node.Searcher.Search(ctx, nodeRequest)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+
+		healthRequest := health.Request{
+			Cluster:   clusterResponse.Status,
+			ClusterID: clusterID,
+			Nodes:     nodeResponse.Nodes,
+		}
+		healthResponse, err := e.service.Health.Searcher.Search(ctx, healthRequest)
```

Comment:
This feels correct to me. Maybe stating the purpose of all this would help in the PR description:

We are trying to separate the concerns of information collection from the concern of health evaluation. Therefor we separated these different concerns into different services.

Reply:
You already did this while I was reviewing :D

Reply:
We're on the same page :)